File: languages/cirq/prime.py
# languages/cirq/prime.py

import cirq

def is_prime(num: int) -> bool:
    if num <= 1:
        return False
    for i in range(2, int(num ** 0.5) + 1):
        if num % i == 0:
            return False
    return True

def main():
    print("Enter a number to check if it's prime: ", end="")
    num = int(input())
    if is_prime(num):
        print(f"{num} is a prime number.")
    else:
        print(f"{num} is not a prime number.")

if __name__ == "__main__":
    main()

File: languages/graphql/prime.js
// languages/graphql/prime.js
const { graphql, buildSchema } = require('graphql');

// Define the schema with a query for prime check
const schema = buildSchema(`
  type Query {
    isPrime(num: Int!): String
  }
`);

// Define the resolver for the isPrime query
const root = {
  isPrime: ({ num }) => {
    if (num <= 1) return `${num} is not a prime number.`;
    for (let i = 2; i <= Math.sqrt(num); i++) {
      if (num % i === 0) return `${num} is not a prime number.`;
    }
    return `${num} is a prime number.`;
  },
};

// Function to run the GraphQL query
const runQuery = async (num) => {
  const query = `{ isPrime(num: ${num}) }`;
  const response = await graphql(schema, query, root);
  console.log(response.data.isPrime);
};

// Read input number from command line
const number = process.argv[2];
runQuery(parseInt(number));
